README.md

```diff
@@ -46,8 +46,14 @@ failing check to be used for testing purposes.
 There are two types of test: the application test and the charm test.
 
 ### Application tests
-To run the application tests, the `GITHUB_TOKEN` environment variable must be set. This
+To run the application tests, the `AUTH_GITHUB_TOKEN` environment variable must be set. This
 should be a token of a user with full repo permissions for the test repository. 
+You can also pass in `AUTH_APP_ID`, `AUTH_INSTALLATION_ID`, and `AUTH_PRIVATE_KEY` 
+to test the authentication using Github App Auth. In that case, the tests will randomly select
+either the token or the app auth to run the tests. Note that the Github app should be installed
+in the test repository organisation/user namespace, with access granted to the test repository
+and in the user namespace that owns the Github token with access to the forked repository.
```

Comment:
Should we have a test run for either and just not execute that run if the environment variable is not set?

Reply:
Updated the code.